scripts/globals/abilities/mijin_gakure.lua

```diff
@@ -25,6 +25,10 @@ function onUseAbility(player,target,ability)
     target:takeDamage(dmg, player, dsp.attackType.SPECIAL, dsp.damageType.ELEMENTAL)
 
     player:setLocalVar("MijinGakure", 1)
+    if (player:getMod(dsp.mod.MIJIN_NAGI) > 0) then
+        player:delStatusEffect(dsp.effect.RERAISE)
+        player:addStatusEffect(dsp.effect.RERAISE,1,0,3600)
```

Comment:
What if you already had a reraise with a higher power than this? Is replacing with rr 1 what is supposed to happen?

Reply:
I searched the youtubes for a reference video of how this is supposed to behave generally, and for the RR effect specifically, without success.

That said, I considered multiple ways to handle it (creating reraise Tier 4, creating a completely separate buff, etc), but without a reference it is what it actually does, it's a just a best guess. I'd like to believe that a reraise that gives you full HP would be at a higher tier of effectiveness than RR1-3, so I planned for it that way.

Alternately, without the removal, if you have any reraise effect active it will show two reraise icons momentarily, but still process the buffed reraise due to nagi first based on the order of operations.

This just seemed like the cleanest way to present it without having a working example.

Reply:
well what you have here would delete my reraise3 and then apply reraise1, and your core check is looking specifically for reraise1 only. Since mijin prevents the exp loss, if it consumes any existing reraise anyway its not going to matter I guess. If it lets reraise status stay then this might get complicated.

We should probably just leave m_hasRaise out of the check entirely, you'll get the result without a blinky status re-add.

Reply:
p.s. you also have to change the line that checks for rr >0 to check for this.

Reply:
I'd also prefer we call this mod MIJIN_RERAISE if since we're not reusing existing mods for equipped item reraise.

(if we do need additional RR tiers, this is where you'd edit for it: https://github.com/DarkstarProject/darkstar/blob/ba4d44dc6039b61c0b217596f9176939930949a9/src/map/entities/charentity.cpp#L1666 and not mess with the status effect)

Reply:
Just noticed the existing mod for mijin appears to have been refactored out and replaced with a localvar.

https://github.com/DarkstarProject/darkstar/commit/bd91bc2583dc03e8007c236d8a9f9eccf9435943

https://github.com/DarkstarProject/darkstar/blob/a110d870d57587c7bcf6ba29f711a1744aa88eec/src/map/modifier.h#L292

so I guess modID 258 can become MIJIN_RERAISE ?